x/concentrated-liquidity/query.go

```diff
@@ -336,3 +336,93 @@ func (k Keeper) GetNumNextInitializedTicks(ctx sdk.Context, poolId, numberOfNext
+// Errors:
+// * ticks given not found
+// * There exists at least 1 tick inside the given range
+// TODO: Currently, we can only get the assets of 2 consecutive ticks
+// need to think of a way to get assets between any 2 ticks
+func (k Keeper) TickRangeUnderlyingAssets(ctx sdk.Context, poolId uint64, lowerTick, upperTick int64) (queryproto.TickRangeUnderlyingAssetsResponse, error) {
+	pool, err := k.getPoolById(ctx, poolId)
+	if err != nil {
+		return queryproto.TickRangeUnderlyingAssetsResponse{}, err
+	}
+
+	// Check if the provided tick range is valid according to the pool's tick spacing and module parameters.
+	if err := validateTickRangeIsValid(pool.GetTickSpacing(), lowerTick, upperTick); err != nil {
+		return queryproto.TickRangeUnderlyingAssetsResponse{}, err
+	}
+
+	// Check lowerTick and upperTick is valid
+	_, err = k.getTickByTickIndex(ctx, poolId, lowerTick)
+	if err != nil {
+		return queryproto.TickRangeUnderlyingAssetsResponse{}, fmt.Errorf("lower tick (%d) not found, error: %w", lowerTick, err)
+	}
+
+	_, err = k.getTickByTickIndex(ctx, poolId, upperTick)
+	if err != nil {
+		return queryproto.TickRangeUnderlyingAssetsResponse{}, fmt.Errorf("upper tick (%d) not found, error: %w", upperTick, err)
+	}
+
+	//--------- Calculate liquidity of tick range -----------
+
+	// Find first tick initialized
+	// use false for zeroForOne since we're going from lower tick -> upper tick
+	zeroForOne := false
+	swapStrategy := swapstrategy.New(zeroForOne, osmomath.ZeroBigDec(), k.storeKey, osmomath.ZeroDec())
+
+	nextTickIter := swapStrategy.InitializeNextTickIterator(ctx, poolId, types.MinCurrentTick)
+	defer nextTickIter.Close()
+
+	if !nextTickIter.Valid() {
+		return queryproto.TickRangeUnderlyingAssetsResponse{}, types.RanOutOfTicksForPoolError{PoolId: poolId}
+	}
+
+	nextTick, err := types.TickIndexFromBytes(nextTickIter.Key())
+	if err != nil {
+		return queryproto.TickRangeUnderlyingAssetsResponse{}, err
+	}
+
+	tick, err := k.getTickByTickIndex(ctx, poolId, nextTick)
+	if err != nil {
+		return queryproto.TickRangeUnderlyingAssetsResponse{}, err
+	}
+
+	// use the smallest tick initialized as the starting point for calculating liquidity.
+	totalLiquidity := tick.LiquidityNet
+	previousTickIndex := nextTick
+	nextTickIter.Next()
+
+	// Loop from first tick unit to unit that input range represent
+	for ; nextTickIter.Valid(); nextTickIter.Next() {
+		tickIndex, err := types.TickIndexFromBytes(nextTickIter.Key())
+		if err != nil {
+			return queryproto.TickRangeUnderlyingAssetsResponse{}, err
+		}
+
+		tickStruct, err := ParseTickFromBz(nextTickIter.Value())
+		if err != nil {
+			return queryproto.TickRangeUnderlyingAssetsResponse{}, err
+		}
+
+		if previousTickIndex == lowerTick && tickIndex == upperTick {
+			break
+		}
+		if previousTickIndex == lowerTick && tickIndex != upperTick {
+			return queryproto.TickRangeUnderlyingAssetsResponse{}, fmt.Errorf("Can not get liquidity of range [%d, %d]", lowerTick, upperTick)
+		}
```

Comment:
do we need this because we are supposedly only supporting consecutive ticks for now?